modules/product/src/main/java/com/opengamma/strata/product/etd/EtdPosition.java

```diff
@@ -21,20 +22,11 @@
  * In many cases, only a long quantity or short quantity will be present with the other set to zero.
  * However it is also possible for both to be non-zero, allowing long and short positions to be treated separately.
  * The net quantity is available via {@link #getQuantity()}.
+ *
+ * @param <T> the type of securitized product
  */
-public interface EtdPosition
-    extends Position {
-
-  /**
-   * Gets the currency of the position.
-   * <p>
-   * This is the currency of the security.
-   *
-   * @return the trading currency
-   */
-  public default Currency getCurrency() {
-    return getSecurity().getCurrency();
-  }
+public interface EtdPosition<T extends SecuritizedProduct>
```

Comment:
Is it necessary to change this interface? Could just make the concrete classes implement `SecuritizedProduct` without changing here, then no need for generics,